### DISCUSSION TITLE
Translating strings in package.json

### DISCUSSION THREAD

**Discussion by ebennett48375** [ivanhofer/typesafe-i18n#161] [Q&A] [answered]
In a VSCode extension, is there any way to use typesafe-i18n to translate strings in package.json? I'm thinking of the text for commands, configuration settings and the like.

**Comment by ivanhofer** [comment, marked as answer]
You would need to write a script that generates the `package.json` file for you. Something like:

```ts
const pck = {
   scripts: {
      test: `ECHO "${LL.hello()}"`
   }
}

writeFile(resolve('./package.json'), JSON.stringify(pck, undefined, 3))
```

Does this work for you?

**Reply by ebennett48375** [reply]
Thank you so much for your speedy reply. I am impressed.

It turns out, however, that VSCode takes care of the issue for you.  If you have a package.nls.json, and, say, package.nls.ja.json, and you use strings like %extension.somekey% in package.json, VSCode automagically does the translation using the json files.
